vowpalwabbit/fb_parser/src/parse_example_flatbuffer.cc

```diff
@@ -79,17 +83,32 @@ bool read_span_flatbuffer(
     sstream << "fb_parser error: flatbuffer size prefix does not match actual size" << std::endl;
     sstream << "   span => @" << std::hex << address << std::dec << " size_prefix = " << flatbuffer_object_size
             << " length = " << length;
-    THROW(sstream.str());
-    return false;
+
+    RETURN_ERROR_LS(status, fb_parser_span_length_mismatch) << sstream.str();
   }
 
   VW::multi_ex temp_ex;
-  temp_ex.push_back(&example_factory());
+  auto scope_guard = VW::scope_exit([&temp_ex, &all, &example_sink]()
+  {
+    if (example_sink == nullptr) { VW::finish_example(*all, temp_ex); }
+    else { example_sink(std::move(temp_ex)); }
+  });
+
+  // There is a bit of unhappiness with the interface of the read_XYZ_<format>() functions, because they often
+  // expect the input multi_ex to have a single "empty" example there. This contributes, in part, to the large
+  // proliferation of entry points into the JSON parser(s). We want to avoid exposing that insofar as possible,
+  // so we will check whether we already received a perfectly good example and use that, or create a new one if
+  // needed.
+  if (examples.size() > 0)
+  {
+    temp_ex.push_back(examples[0]);
+    examples.pop_back();
+  }
+  else { temp_ex.push_back(&example_factory()); }
```

Comment:
It looks like the first element from examples is being added to temp_ex but last element is removed from examples.  Is this intentional?

Reply:
The expectation here is that elements[0] is also the last element. May be worth to add an assert here.